kie-server-parent/kie-server-controller/kie-server-controller-impl/src/test/java/org/kie/server/controller/impl/storage/ServerTemplateStorageTest.java

```diff
@@ -0,0 +1,371 @@
+
+        ServerInstanceKey serverInstanceKey = instances.iterator().next();
+        assertNotNull(serverInstanceKey);
+        assertEquals(serverTemplate.getId(), serverInstanceKey.getServerTemplateId());
+        if (isRequired()) {
+            assertEquals(serverTemplate.getId() + "@localhost:8080", serverInstanceKey.getServerName());
+            assertEquals(serverTemplate.getId() + "@localhost:8080", serverInstanceKey.getServerInstanceId());
+            assertEquals("http://localhost:8080/server", serverInstanceKey.getUrl());
+        }
+
+        Collection<ContainerSpec> containerSpecs = template.getContainersSpec();
+        assertNotNull(containerSpecs);
+        assertEquals(2, containerSpecs.size());
+
+        Iterator<ContainerSpec> iterator = containerSpecs.iterator();
+
+        // first container spec...
+        ContainerSpec spec = iterator.next();
+        assertNotNull(spec);
+
+        assertEquals(containerSpec.getId(), spec.getId());
+        assertEquals(containerSpec.getReleasedId(), spec.getReleasedId());
+        assertEquals(containerSpec.getServerTemplateKey().getId(), spec.getServerTemplateKey().getId());
+        if (isRequired()) {
+            assertEquals(containerSpec.getServerTemplateKey().getName(), spec.getServerTemplateKey().getName());
+        }
+        assertEquals(containerSpec.getConfigs().size(), spec.getConfigs().size());
+
+        assertTrue(spec.getConfigs().containsKey(Capability.RULE));
+        assertTrue(spec.getConfigs().containsKey(Capability.PROCESS));
+
+        RuleConfig ruleConfig = (RuleConfig) spec.getConfigs().get(Capability.RULE);
+        assertNotNull(ruleConfig);
+        assertEquals(this.ruleConfig.getPollInterval(), ruleConfig.getPollInterval());
+        assertEquals(this.ruleConfig.getScannerStatus(), ruleConfig.getScannerStatus());
+
+        ProcessConfig processConfig = (ProcessConfig) spec.getConfigs().get(Capability.PROCESS);
+        assertNotNull(processConfig);
+        assertEquals(this.processConfig.getKBase(), processConfig.getKBase());
+        assertEquals(this.processConfig.getKSession(), processConfig.getKSession());
+        assertEquals(this.processConfig.getMergeMode(), processConfig.getMergeMode());
+        assertEquals(this.processConfig.getRuntimeStrategy(), processConfig.getRuntimeStrategy());
+
+        // second container spec
+        spec = iterator.next();
+        assertNotNull(spec);
+
+        assertEquals(newContainerSpec.getId(), spec.getId());
+        assertEquals(newContainerSpec.getReleasedId(), spec.getReleasedId());
+        assertEquals(newContainerSpec.getServerTemplateKey().getId(), spec.getServerTemplateKey().getId());
+        if (isRequired()) {
+            assertEquals(newContainerSpec.getServerTemplateKey().getName(), spec.getServerTemplateKey().getName());
+        }
+        assertEquals(newContainerSpec.getConfigs().size(), spec.getConfigs().size());
+    }
+
+    protected void createServerTemplateWithContainer() {
```

Comment:
Is there any reason why this method is called in implementation classes and not here in `@Before` method?

Reply:
That method is called by concrete test case class at the @Before method. For example,

https://github.com/rhtevan/droolsjbpm-integration/blob/fd4635e8e9b2263458db3339d7ea0c94b7ae87b6/kie-server-parent/kie-server-controller/kie-server-controller-openshift/src/test/java/org/kie/server/controller/openshift/storage/OpenShiftServerTemplateStorageTest.java#L102